google/reader.go

```diff
@@ -54,16 +54,18 @@ func (r *GCPReader) ListInstances(filter string) (map[string][]compute.Instance,
 		return nil, err
 	}
 	for _, zone := range zones {
-		instances := make([]compute.Instance, 0)
-		il, err := is.List(r.project, zone).
+		// 500 is the current `maxResults`
+		instances := make([]compute.Instance, 0, 500)
```

Comment:
And if you have more than 500? It's not possible to request them all?

Reply:
I use pages. Each page has a default size of 500 elements: see [pagination](https://godoc.org/google.golang.org/api/compute/v1#InstancesListCall.Pages) and [max result](https://godoc.org/google.golang.org/api/compute/v1#InstancesListCall.MaxResults)

`instances` slice will store only the instances returned by the current page (so 500 maximum). Then we append this slice to the final `instancesList` map.